spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -221,7 +250,8 @@ public void addTopic(String topic, int qos) {
 		try {
 			super.addTopic(topic, qos);
 			if (this.mqttClient != null && this.mqttClient.isConnected()) {
-				this.mqttClient.subscribe(topic, qos).waitForCompletion(getCompletionTimeout());
+				this.mqttClient.subscribe(new MqttSubscription(topic, qos), this::messageArrived)
```

Comment:
Don't we need to subscribe to newly added topics even if the client comes from the manager?

Reply:
We init `mqttClient` field from the client manager in the start method, just as for v3 adapter 🙂 
So this code is almost not changed, at least no additional `if`s